src/main.rs

```diff
@@ -1376,13 +1377,126 @@ fn parse_manpage_at_path(
 #[derive(Copy, Clone, Debug)]
 struct Progress(pub bool);
 
-// TODO Arg/output types?
-fn parse_and_output_man_pages(
-    _paths: impl Iterator<Item = PathBuf>,
-    _output_directory: PathBuf,
-    Progress(_show_progress): Progress,
-) {
-    unimplemented!();
+// Output: Result<cmd_name, error>
+// TODO: Result<cmd_name, CompletionsError>
+fn parse_and_output_man_pages<P: AsRef<Path>>(
+    paths: impl Iterator<Item = P>,
+    output_directory: P,
+    Progress(show_progress): Progress,
+    deroff_only: bool,
+    write_to_stdout: bool,
+) -> Result<(), String> {
+    let mut paths = paths
+        .map(|path| path.as_ref().to_owned())
+        .collect::<Vec<_>>();
+    paths.sort();
+    paths.dedup();
+
+    let total_count = paths.len();
+
+    let (mut successful_count, mut index) = (0, 0);
+
+    let mut cmd_name = "".to_owned();
+
+    // Get the number of digits in total_count
+    let padding_len = (total_count as f32).log(10.) as usize + 1;
```

Comment:
Err, what is this? Did you mean `total_count.to_string().len()`?

Reply:
@charliethomson Is this incorrect?

Reply:
Not sure if log is faster, but you could do `log10()`.

Reply:
Log is O(1), `to_string().len()` is O(2N)
I didn't know log10 was a thing. I'll swap that real quick

Reply:
Ah, I didn't know about that. This method does seemed interesting where do you find this? I tried `ceil`, that does not work, seemed like we need `+ 1`.

Reply:
:shrug: one of my friends is really good at that type of thing, she let me know about it a few months back. Pretty good solution tbh